### DISCUSSION TITLE
Microcks SSRF Vulnerability

### DISCUSSION THREAD

**Discussion by b33t1e** [microcks/microcks#892] [Show and tell]
Microcks up to 1.17.1 was discovered to contain a Server-Side Request Forgery (SSRF) via the component /jobs and /artifact/download. This vulnerability allows attackers to access network resources and sensitive information via a crafted GET request.
<img width="903" alt="2" src="https://github.com/microcks/microcks/assets/33768821/f304ebd3-1767-49e9-8851-3798f03c1703">
<img width="1020" alt="3" src="https://github.com/microcks/microcks/assets/33768821/26f18d3b-646a-4799-a37a-e16371a49823">

**Comment by lbroudoux** [comment]
Hello,

Thanks for raising this discussion. I have to admit I'm lacking a bit of context on this topic as I don't know how to evaluate the risks and the impact of such a vulnerability. What can typically do an attacker that would exploit it?

I mean I have basic knowledge of SSRF but in this particular use case how could we avoid connecting to an external URL? And is there some risk as we're expecting a well-formed artifact and not running arbitrary code that may be provided by this external URL? Thanks for your explanation.

**Reply by b33t1e** [reply]
I apologize for the late response. This SSRF could pose a threat to the internal network resources where Microcks is located. For example, if there is a vulnerable web service within the internal network where Microcks resides, an attacker could exploit this SSRF vulnerability to send HTTP requests and carry out an attack, thereby compromising the overall security of the internal network.

The recommended fix provided here is to filter internal network IP ranges, such as 127.0.0.1, 192.168.0.1/24, 172.16.0.1/24, etc., in order to protect the security of the internal network resources.
